### PR TITLE
feat(clerk-js): Add dev mode warning to components (#3511)

### PR DESCRIPTION
**Pull request by octoper**
## Description

Adding a development mode warning to our components to avoid going to productions with development keys by accident.

To be able to disable the development mode warning in order to preview how the components will look like in production the `appearance.layout.unsafe_disableDevelopmentModeWarnings` key has been added.

### Example usage:

**Globally using the `<ClerkProvider />`:**
```typescript
<ClerkProvider appearance={{
 layout: {
   unsafe_disableDevelopmentModeWarnings: true,
  }
}}>
...
</ClerkProvider>
```

**On each component:**
```typescript
<UserProfile appearance={{
 layout: {
   unsafe_disableDevelopmentModeWarnings: true,
  }
}} />
```

Original PR: #3511 

### Screenshots

CleanShot 2024-07-25 at 15 30 12@2x

CleanShot 2024-07-12 at 03 22 47@2x

CleanShot 2024-07-25 at 15 31 12@2x

CleanShot 2024-07-31 at 13 08 00@2x

CleanShot 2024-07-31 at 13 08 33@2x

## Checklist

- [x] `npm test` runs as expected.
- [x] `npm run build` runs as expected.
- [ ] (If applicable) [JSDoc comments](https://jsdoc.app/about-getting-started.html) have been added or updated for any package exports
- [ ] (If applicable) [Documentation](https://github.com/clerk/clerk-docs) has been updated

## Type of change

- [ ] 🐛 Bug fix
- [x] 🌟 New feature
- [ ] 🔨 Breaking change
- [ ] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

### CODE REVIEW

**Comment by octoper** [comment]
!preview

**Comment by clerk-cookie** [comment]
Hey @octoper, your preview is currently deploying and should be available shortly.

| Status | Preview | Updated (UTC) |
| :-- | :-- | :-- |
| :cook: _Deploying..._ | [Inspect](https://javascript-preview-r3m2q6tcy.clerkstage.dev) | Aug 01, 2024 08:54 AM |

**Review by anagstef — APPROVED** [review]

File: packages/clerk-js/src/core/resources/DisplayConfig.ts [hunk 1]
```diff
@@ -80,6 +81,7 @@ export class DisplayConfig extends BaseResource implements DisplayConfigResource
     this.afterLeaveOrganizationUrl = data.after_leave_organization_url;
     this.afterCreateOrganizationUrl = data.after_create_organization_url;
     this.googleOneTapClientId = data.google_one_tap_client_id;
+    this.showDevModeWarning = data.show_devmode_warning;
```
Comment (thiskevinwang) [inline packages/clerk-js/src/core/resources/DisplayConfig.ts:84]: 👍

**Comment by octoper** [comment]
!preview

**Comment by octoper** [comment]
!snapshot

**Comment by clerk-cookie** [comment]
Hey @octoper - the snapshot version command generated the following package versions:
| Package | Version |
| --- | --- |
| @clerk/astro | 1.0.5-snapshot.v741fdbd |
| @clerk/backend | 1.5.0-snapshot.v741fdbd |
| @clerk/chrome-extension | 1.1.8-snapshot.v741fdbd |
| @clerk/clerk-js | 5.11.0-snapshot.v741fdbd |
| @clerk/elements | 0.11.0-snapshot.v741fdbd |
| @clerk/clerk-expo | 2.0.1-snapshot.v741fdbd |
| @clerk/express | 0.0.22-snapshot.v741fdbd |
| @clerk/fastify | 1.0.24-snapshot.v741fdbd |
| gatsby-plugin-clerk | 5.0.0-beta.45 |
| @clerk/localizations | 2.5.3-snapshot.v741fdbd |
| @clerk/nextjs | 5.2.9-snapshot.v741fdbd |
| @clerk/clerk-react | 5.3.0-snapshot.v741fdbd |
| @clerk/remix | 4.2.8-snapshot.v741fdbd |
| @clerk/clerk-sdk-node | 5.0.21-snapshot.v741fdbd |
| @clerk/shared | 2.4.2-snapshot.v741fdbd |
| @clerk/tanstack-start | 0.1.12-snapshot.v741fdbd |
| @clerk/testing | 1.2.4-snapshot.v741fdbd |
| @clerk/themes | 2.1.15-snapshot.v741fdbd |
| @clerk/types | 4.10.0-snapshot.v741fdbd |

Tip: Use the snippet copy button below to quickly install the required packages.
`@clerk/astro`
```sh
npm i @clerk/astro@1.0.5-snapshot.v741fdbd --save-exact
```
`@clerk/backend`
```sh
npm i @clerk/backend@1.5.0-snapshot.v741fdbd --save-exact
```
`@clerk/chrome-extension`
```sh
npm i @clerk/chrome-extension@1.1.8-snapshot.v741fdbd --save-exact
```
`@clerk/clerk-js`
```sh
npm i @clerk/clerk-js@5.11.0-snapshot.v741fdbd --save-exact
```
`@clerk/elements`
```sh
npm i @clerk/elements@0.11.0-snapshot.v741fdbd --save-exact
```
`@clerk/clerk-expo`
```sh
npm i @clerk/clerk-expo@2.0.1-snapshot.v741fdbd --save-exact
```
`@clerk/express`
```sh
npm i @clerk/express@0.0.22-snapshot.v741fdbd --save-exact
```
`@clerk/fastify`
```sh
npm i @clerk/fastify@1.0.24-snapshot.v741fdbd --save-exact
```
`gatsby-plugin-clerk`
```sh
npm i gatsby-plugin-clerk@5.0.0-beta.45 --save-exact
```
`@clerk/localizations`
```sh
npm i @clerk/localizations@2.5.3-snapshot.v741fdbd --save-exact
```
`@clerk/nextjs`
```sh
npm i @clerk/nextjs@5.2.9-snapshot.v741fdbd --save-exact
```
`@clerk/clerk-react`
```sh
npm i @clerk/clerk-react@5.3.0-snapshot.v741fdbd --save-exact
```
`@clerk/remix`
```sh
npm i @clerk/remix@4.2.8-snapshot.v741fdbd --save-exact
```
`@clerk/clerk-sdk-node`
```sh
npm i @clerk/clerk-sdk-node@5.0.21-snapshot.v741fdbd --save-exact
```
`@clerk/shared`
```sh
npm i @clerk/shared@2.4.2-snapshot.v741fdbd --save-exact
```
`@clerk/tanstack-start`
```sh
npm i @clerk/tanstack-start@0.1.12-snapshot.v741fdbd --save-exact
```
`@clerk/testing`
```sh
npm i @clerk/testing@1.2.4-snapshot.v741fdbd --save-exact
```
`@clerk/themes`
```sh
npm i @clerk/themes@2.1.15-snapshot.v741fdbd --save-exact
```
`@clerk/types`
```sh
npm i @clerk/types@4.10.0-snapshot.v741fdbd --save-exact
```

**Review by alexcarpenter — APPROVED** [review]

**Pull request merged**